src/lib/y2network/udev_rule.rb

```diff
@@ -53,7 +53,15 @@ def find_for(device)
       # @param mac  [String] MAC address
       def new_mac_based_rename(name, mac)
         new_network_rule(
-          [UdevRulePart.new("ATTR{address}", "=", mac), UdevRulePart.new("NAME", "=", name)]
+          [
+            # Guard to not try to rename everything with the same MAC address (e.g. vlan devices
+            # inherit the MAC address from the underlying device).
+            UdevRulePart.new("KERNEL", "==", "eth*"),
```

Comment:
well, problem with that rule is that it works only for Ethernet. For WIFI this won't work. And I worry also for IB and other types.

Reply:
So guard is good, but kernel name depends on interface type.

Reply:
And not only that: we have detected that it won't work when using predictable names (for instance, think about openSUSE). So after discussing with @teclator, we have decided to keep the old behaviour and add a new PBI to do some research about the best way to handle it.